### PR TITLE
(TWILL-136) Override equals and hashCode for JvmOptions.DebugOptions to test equality

### PR DESCRIPTION
**Pull request by hsaputra**
The current code for JvmOptions.DebugOptions does not have equals and hashCode overriden for equality test.

This would cause fail comparison for DebugOptions.NO_DEBUG when being used in YarnTwillPreparer:

```
final class YarnTwillPreparer implements TwillPreparer {

...

  @Override
  public TwillPreparer enableDebugging(boolean doSuspend, String... runnables) {
    this.debugOptions = new JvmOptions.DebugOptions(true, doSuspend, ImmutableSet.copyOf(runnables));
    return this;
  }

....
  private void saveJvmOptions(Map<String, LocalFile> localFiles) throws IOException {
    if ((extraOptions == null || extraOptions.isEmpty()) &&
      JvmOptions.DebugOptions.NO_DEBUG.equals(this.debugOptions)) {
      // If no vm options, no need to localize the file.
      return;
    }
    ...
  }
...
}
```

### CODE REVIEW

**Comment by hsaputra** [comment]
Weird test fails, close it to fix the PR

File: twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java [hunk 1]
```diff
@@ -83,5 +84,28 @@ public boolean doSuspend() {
     public boolean doDebug(String runnable) {
       return doDebug && (runnables == null || runnables.contains(runnable));
     }
+
+    @Override
+    public boolean equals(Object object) {
+      if (this == object) {
+        return true;
+      }
+      if (object == null || !(object instanceof DebugOptions)) {
```
Comment (keith-turner) [inline twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java]: I don't think null check is needed.

http://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

**Comment by hsaputra** [comment]
Cannot reopen due to force update, please check #43

File: twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java (hunk 1 above)
Comment (hsaputra) [inline twill-core/src/main/java/org/apache/twill/internal/JvmOptions.java]: Thanks for the review, I moved the PR to #43

**Pull request closed without merging**